packages/grid/src/GridMetricCalculator.ts

```diff
@@ -1755,7 +1753,9 @@ class GridMetricCalculator {
    * @param row The row model index to reset
    */
   resetRowHeight(row: ModelIndex): void {
-    this.setRowHeight(row, undefined);
+    const userRowHeights = new Map(this.userRowHeights);
```

Comment:
Curious why the reset and set height/width methods create a new map. If it were in a component then it would be needed to trigger update lifecycles, but not sure why in this case. If there's a reason, we should document it so that somebody in the future doesn't come through and remove it as an unnecessary clone

Reply:
Yea it's because a consumer of the metrics may memoize based on column widths/row heights (like we do when saving the state in IrisGridPanel). Otherwise there's no quick way to tell if the values have changed/stayed the same